[PEAUTY-80] ReDesign Designer Domain & develop Designer CRUD

harvartz opened the pull request:
## 📄 [PEAUTY-80] ReDesign Designer Domain & develop Designer CRUD

Jira : [PEAUTY-80](url)

## ✨ 변경 사항
- [x] 기능 추가/변경 설명

<!-- Pull Request의 설명을 추가하세요. -->
- 변경된 요구사항에 맞춰서 designer, workspace, rating, badge, license 도메인을 재설계 했습니다.
- 엔티티에 대한 연관관계를 없애고 fk를 직접 대입했습니다.
- 디자이너 Create, Read API에 대한 작업을 진행했습니다. 
- 디자이너 Update 작업 완료했습니다.

## 📅 작업 일정
<!-- 해당 작업을 수행하는데 예상했던 공수와 실제 소요되었던 공수를 기입해주세요. -->
- Expected MD: 4.5MD
- Actual MD: 4.5MD
### Difference reason (If correct, no need.)
- None.

## ✔️ 확인 사항

- [x] 코드가 잘 작동하는지 확인했나요?
- [x] 새로운 기능에 대한 테스트가 추가되었나요?
- [x] 문서가 업데이트되었나요?

harvartz committed:
feat: entity 재설계
- 바뀐 ERD에 맞춰서 entity 재설계

harvartz committed:
style: 사용하지 않은 import 구문 삭제

harvartz committed:
fix: 미용사 도메인 재설계
- shop + designerInfo -> workspace 도메인으로 통합
- license 도메인 추가
- workspace 도메인 안에 license 도메인 위치

harvartz committed:
fix: 미용사 도메인 및 엔티티 2차 재설계
[도메인]
- designer - workspace 분리
- designer 안에 license 종속
- workspace 안에 rating 종속
[엔티티]
- rating - designer 연관관계
- 연관관계 -> FK를 Long Id 값으로 직접 입력

harvartz committed:
feat: 미용사 워크 스페이스 Create API 개발
[TODO]
- Response로 Badge와 Scissors는 어떻게 할 것인가?
- Enum 타입인 Payment는 테이블로 따로 나눌 것인가?
- H2는 Enum 타입을 지원하지 않아 Payment는 데이터를 null로 하여 체크
- 배너 이미지도 리스트로 받는다는 요구사항이 추가되었음

harvartz committed:
style: 사용하지 않은 import 구문 제거

harvartz committed:
feat: rating, badge, designerBadge 도메인, 엔티티 설계

harvartz committed:
style: 사용하지 않는 import 삭제

harvartz committed:
style: 사용하지 않는 import, 파일 삭제

Shanate commented on peauty-designer-api/src/main/java/com/peauty/designer/business/designer/DesignerPort.java:
```diff
@@ -7,10 +7,17 @@
 
 public interface DesignerPort {
     void checkCustomerNicknameDuplicated(String nickname);
```
메서드 이름을 변경해야할 것 같습니다. Designer 포트인데 Customer라고 되어있네요

harvartz replied:
크흠 수정하겠습니다.. : )

Shanate commented on peauty-designer-api/src/main/java/com/peauty/designer/implementation/designer/DesignerAdapter.java:
```diff
@@ -63,15 +73,45 @@ public Designer registerNewDesigner(SignUpCommand command) {
+                        .badgeName(badgeEntity.getBadgeName())
+                        .badgeContent(badgeEntity.getBadgeContent())
+                        .badgeImageUrl(badgeEntity.getBadgeImageUrl())
+                        .isRepresentativeBadge(true)
```
이게 대표 뱃지만 보여주겠다는 것을 의미하는건가요?

harvartz replied:
맞습니다. 조회 화면에 대표 뱃지에 관련된 정보만 있어서 isRepresentativeBadge 상태 값이 true인 것만 JPQL로 조회했습니다 : )

Shanate commented on peauty-designer-api/src/main/java/com/peauty/designer/implementation/designer/DesignerAdapter.java:
```diff
@@ -63,15 +73,45 @@ public Designer registerNewDesigner(SignUpCommand command) {
+                        .badgeImageUrl(badgeEntity.getBadgeImageUrl())
+                        .isRepresentativeBadge(true)
+                        .build())
+                .collect(Collectors.toList());
```
라이센스를 조회하는 것과 뱃지를 조회하는 것에 대해서 구두로 한 번 설명 부탁드려도 될까요

Shanate reviewed (approved):
이것저것 물리는게 많아서 더 고생하셨을게 보입니다. 수고하셨어요 설명 필요한 부분은 코멘트 남겨놨습니다.

harvartz committed:
feat: 디자이너 update API 개발

harvartz committed:
style: 사용하지 않은 import 구문 삭제ㅔ

harvartz committed:
Merge branch 'release/v0.1.0' into feature/PEAUTY-80

harvartz committed:
fix: phoneNum -> phoneNumber 수정

harvartz committed:
refactor: customer -> designer
- checkDesignerNicknameDuplicated -> checkDesignerNicknameDuplicated
- checkDesginerPhoneNumDuplicated -> checkDesignerPhoneNumDuplicated

weejinyoung commented on peauty-designer-api/src/main/java/com/peauty/designer/presentation/controller/auth/AuthController.java:
```diff
@@ -61,7 +61,7 @@ private void redirectBySignInResult(HttpServletResponse response, SignInResult r
                     .path("/signup")
                     .queryParam("socialId", result.socialId())
                     .queryParam("socialPlatform", result.socialPlatform())
-                    .queryParam("nickname", result.nickname())
+                    .queryParam("workspaceName", result.nickname())
                     .queryParam("profileImageUrl", result.profileImageUrl())
```
rollback 해주세요 11

weejinyoung commented on peauty-designer-api/src/main/java/com/peauty/designer/presentation/controller/auth/AuthTestController.java:
```diff
@@ -50,7 +50,7 @@ public PeautyApiResponse<String> testSignUp(
         String response = String.format("""
            socialId: %s
            socialPlatform: %s
-           nickname: %s
+           workspaceName: %s
            profileImageUrl: %s
```
rollback 해주세요 22

weejinyoung commented on peauty-persistence/src/main/java/com/peauty/persistence/designer/WorkspaceEntity.java:
```diff
@@ -0,0 +1,66 @@
+
+    @Column(name = "open_days", length = 10, nullable = false)
+    private String openDays;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "payment_options", length = 255, nullable = false)
+    private List<PaymentOption> paymentOptions;
+
```
Payment Option 의 List 를 테이블에 저장할 때 어떻게 되는지 주의가 필요해보여용

harvartz replied:
@ElementCollection(fetch = FetchType.LAZY)
 따로 엔티티를 생성하지 않고 어노테이션을 사용하여 DB에 테이블을 생성할 수 있게 진행했습니다! enum 타입이 세 가지 밖에 없고 데이터가 많지 않을 것 같아서 시도했습니다!

weejinyoung reviewed (changes requested):
rollback 요청 확인해주세요
오랜 시간 고민하시느라 고생하셨습니다 🙇

harvartz committed:
fix: enum 타입 테이블 없이 수정
- 실제 테이블을 만들지 않고 @ElementCollection(fetch = FetchType.LAZY)를 사용하여 따로 entity를 만들지 않고 DB상에서만 테이블을 만들어 저장할 수 있게 진행했습니다.

harvartz committed:
refactor: shop -> workspace 폴더명 변경

harvartz committed:
refactor: workspace -> nickname 변경
- 잘못 변경한 변수명 다시 수정

harvartz committed:
docs: swagger 어노테이션 수정

harvartz committed:
fix: create api 수정

harvartz committed:
feat: enum 타입 유효성 검사 추가
